aggregate/boost_test_parser.rb

```diff
@@ -3,47 +3,121 @@
 require_relative 'test_parser'
 
 class BoostTestParser < TestParser
-  def initialize(fn)
-    @doc = REXML::Document.new(File.read(fn))
+  def initialize(fn, glob = false)
+    @docs =
+      if glob
+        Dir.glob(fn).map { |x|
+          REXML::Document.new(File.read(x))
+        }
+      else
+        [REXML::Document.new(File.read(fn))]
+      end
   end
 
   def each_test
-    @doc.root.elements.each('TestSuite') { |ts|
-      ts.elements.each('TestCase') { |tc|
-        name = tc.attribute('name').value
+    @docs.each do |doc|
+      r = doc.root
+      next unless r
+
+      r.elements.each('TestSuite') { |ts|
+        ts.elements.each('TestCase') { |tc|
+          name = tc.attribute('name').value
+
+          raise "Unable to parse name: \"#{name}\"" unless name =~ /^test_(.*?)$/
+          name = underscore_to_ucamelcase($1)
+
+          failures = []
+          tc.elements.each('Error') { |err|
+            failures << TestResult::Failure.new(
+              err.attribute('file'),
+              err.attribute('line'),
+              err.text,
+              nil
+            )
+          }
+          tc.elements.each('Exception') { |err|
+            failures << TestResult::Failure.new(
+              err.attribute('file'),
+              err.attribute('line'),
+              err.text,
+              nil
+            )
+          }
 
-        raise "Unable to parse name: \"#{name}\"" unless name =~ /^test_(.*?)$/
-        name = underscore_to_ucamelcase($1)
-
-        failures = []
-        tc.elements.each('Error') { |err|
-          failures << TestResult::Failure.new(
-            err.attribute('file'),
-            err.attribute('line'),
-            err.text,
-            nil
-          )
+          if failures.empty?
+            status = :passed
+            failure = nil
+          else
+            status = :failed
+            failure = failures[0]
+          end
+
+          # `<TestCase>` elements of tests that are disabled (using the
+          # `--run_test=!<test_name>` command-line option) because they aborted
+          # (typically due to a memory access violation, which terminates the
+          # entire test suite) in a previous run attempt look like this in the
+          # XML log:
+          #
+          # ```xml
+          #     <TestCase name="test_debug_array_user" skipped="yes" reason="disabled"/>
+          # ```
+          #
+          # As you can see, they don't contain any `<TestingTime>` tag, so we
+          # need to make sure we handle this case gracefully.
+          testing_time_node = tc.elements['TestingTime']
+          testing_time =
+            if testing_time_node.nil?
+              0.0
+            else
+              testing_time_node.text.to_f
+            end
```

Comment:
Technically, if we don't mind having nulls in `testing_time`, this will be a tiny bit more concise way to do it:

```suggestion
testing_time = tc.elements['TestingTime']&.text&.to_f
```

or, if we absolutely need it to be `0.0`, then

```suggestion
testing_time = tc.elements['TestingTime']&.text&.to_f || 0.0
```

Not sure about readability of it, though.

Reply:
> Not sure about readability of it, though.

I know about this operator, but yeah, I don't think it's very readable. Maybe

```rb
tc.elements['TestingTime'] && tc.elements['TestingTime'].text.to_f
```

... would be better.

Either way, I don't see much value in this, given that it works the same as the original version which is well readable though somewhat verbose. But if you want, feel free to refactor.

Just noting that the `&.` operator was introduced in Ruby 2.3, so I'm not 100% sure that using it will not break the `ruby/1.9-linux-x86_64` target that we still have (although it's a `ruby` target and not `cpp_stl`, it will still run `aggregate/convert_to_json` like any other language, which [requires `boost_test_parser`](https://github.com/kaitai-io/kaitai_struct_tests/blob/ac7dd7f068a5ea8e3fb86b2ec9666ff12aeb5b76/aggregate/convert_to_json#L12) and I don't know whether merely the presence of `&.` in the source code wouldn't be a syntax error in Ruby 1.9).

Reply:
Yep, agree. Forgot about that Ruby 1.9 aspect completely.